src/MongoDB.Driver/Encryption/AutoEncryptionLibMongoController.cs

```diff
@@ -28,26 +28,61 @@ namespace MongoDB.Driver.Encryption
 {
     internal sealed class AutoEncryptionLibMongoCryptController : LibMongoCryptControllerBase, IBinaryDocumentFieldDecryptor, IBinaryCommandFieldEncryptor
     {
+        #region static
+        public static AutoEncryptionLibMongoCryptController Create(IMongoClient client, CryptClient cryptClient, AutoEncryptionOptions autoEncryptionOptions)
+        {
+            var lazyInternalClient = new Lazy<IMongoClient>(() => CreateInternalClient());
+            var keyVaultClient = autoEncryptionOptions.KeyVaultClient ?? lazyInternalClient.Value;
+            var metadataClient = autoEncryptionOptions.BypassAutoEncryption ? null : lazyInternalClient.Value;
+            var internalClient = lazyInternalClient.IsValueCreated ? lazyInternalClient.Value : null;
+
+            return new AutoEncryptionLibMongoCryptController(
+                internalClient,
+                keyVaultClient,
+                metadataClient,
+                cryptClient,
+                autoEncryptionOptions);
+
+            IMongoClient CreateInternalClient()
+            {
+                var internalClientSettings = client.Settings.Clone();
+                internalClientSettings.AutoEncryptionOptions = null;
+                internalClientSettings.MinConnectionPoolSize = 0;
+                return new MongoClient(internalClientSettings);
+            }
```

Comment:
One more suggestion, which you can apply at your own discretion, because it makes a bit more checks but gets rid of `Lazy`:
```suggestion
            IMongoClient internalClient = null;
            if (autoEncryptionOptions.KeyVaultClient == null || autoEncryptionOptions.BypassAutoEncryption == false)
            {
                var internalClientSettings = client.Settings.Clone();
                internalClientSettings.AutoEncryptionOptions = null;
                internalClientSettings.MinConnectionPoolSize = 0;
                internalClient = new MongoClient(internalClientSettings);
            }
            var keyVaultClient = autoEncryptionOptions.KeyVaultClient ?? internalClient;
            var metadataClient = autoEncryptionOptions.BypassAutoEncryption ? null : internalClient;

            return new AutoEncryptionLibMongoCryptController(
                internalClient,
                keyVaultClient,
                metadataClient,
                cryptClient,
                autoEncryptionOptions);
```

Reply:
I think a lazy approach exactly matches to what is required by the spec. Also, I think the code written in this way is a bit more clear, so I would leave this as is

Reply:
>I don't think the spec requires the use of the Lazy<T> class.

I meant not .NET lazy, but lazy approach in general when we postpone calling a method until it's actually required